libcontainer/cgroups/fs/freezer.go

```diff
@@ -129,7 +129,25 @@ func (s *FreezerGroup) GetState(path string) (configs.FreezerState, error) {
 		case "THAWED":
 			return configs.Thawed, nil
 		case "FROZEN":
-			return configs.Frozen, nil
+			// Check freezer state for current control group to check
+			// if it is frozen, or if it is one of its ancestors
+			self, err := cgroups.ReadFile(path, "freezer.self_freezing")
+			if err != nil {
+				// If the kernel is too old, then we just treat
+				// it as being frozen
+				if os.IsNotExist(err) || errors.Is(err, unix.ENODEV) {
```

Comment:
Formally, new code should use `errors.Is(err, os.IsNotExist)` but either way is fine.